src/graphics/material.cpp

```diff
@@ -60,13 +60,28 @@ Material::Material(const XMLNode *node, bool deprecated)
     m_installed = false;
 
     m_sampler_path[0] = "unicolor_white";
-    node->get("name",      &m_texname);
-    if (m_texname=="")
+    node->get("name",      &m_matname);
+    if (m_matname=="")
     {
-        throw std::runtime_error("[Material] No texture name specified "
+        throw std::runtime_error("[Material] No material name specified "
                                  "in file\n");
     }
 
+    node->get("tex-layer-0",      &m_texname);
+    if (m_texname=="")
```

Comment:
These two lines could be replaced by : 
if (!node->get("tex-layer-0", &m_texname))

It should not be a functional change only a coding style change, using a more compact style already present in other files.